test/box-luatest/gh_7904_export_box_schema_version_to_public_api_and_lua_test.lua

```diff
@@ -0,0 +1,26 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new{
+        alias   = 'dflt',
+    }
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+--[[
+Checks that `box.schema_version` returns a valid number.
```

Comment:
Let's also check that it grows with DDL. Also, you should add a C module API test.

Reply:
> Also, you should add a C module API test.

Added a check to `test/app-tap/module_api.test.lua`.

Reply:
TBH, I'm kind of lazy to add checks for when the returned schema version is supposed to be 0 or shouldn't change.

Reply:
> This is already checked in other tests

Then I don't see any point in this test. Better drop it.

> Added a check to test/app-tap/module_api.test.lua.

The test doesn't belong in `app-tap`, because `box_schema_version` need `box.cfg` to work. Anyway, I agree with @Totktonada that we should also check it more extensively - at least, check that it's incremented on DDL.

Reply:
> Then I don't see any point in this test. Better drop it.

Then what’s the point of testing the C API?

Reply:
> at least, check that it's incremented on DDL.

I already wrote (see comment above) that we have such checks, not sure what you mean here.

Reply:
If the Lua API is tested, it doesn't mean that the C API doesn't have to be. For example, we have the `key_def` module available both in Lua and in C and we test both (`box-tap/key_def.test.lua`, `app-tap/module_api.test.lua`).

Reply:
> Then I don't see any point in this test. Better drop it.

So then why shouldn't the **Lua** API be tested?

Reply:
> So then why shouldn't the **Lua** API be tested?

I didn't mean it shouldn't be tested. What I meant to say is that, since the Lua API is already tested by `box/schema_reload.test.lua` and your test doesn't improve on the existing test, there's no point in it.

Reply:
> The test doesn't belong in `app-tap`, because `box_schema_version` need `box.cfg` to work.

As far as I can see from the test, it already uses `box.cfg`:
https://github.com/tarantool/tarantool/blob/a62e14a0106379b9cfd5198d9826e8d84c917827/test/app-tap/module_api.test.lua#L491-L498

Reply:
> If the Lua API is tested, it doesn't mean that the C API doesn't have to be. For example, we have the `key_def` module available both in Lua and in C and we test both (`box-tap/key_def.test.lua`, `app-tap/module_api.test.lua`).

On the contrary, we have, box API functions, such as `box_txn_set_timeout` or `box_txn_set_isolation` that are not tested.

Reply:
> I didn't mean it shouldn't be tested. What I meant to say is that, since the Lua API is already tested by `box/schema_reload.test.lua` and your test doesn't improve on the existing test, there's no point in it.

So what value would test of the C API add except for checking that this function is indeed exported?

Reply:
> As far as I can see from the test, it already uses `box.cfg`

Too bad, but then I guess it's okay to add one more test case in there that depends on `box.cfg`.

Reply:
> Too bad, but then I guess it's okay to add one more test case in there that depends on box.cfg.

Okay, did just so.